src/Input/Input.js

```diff
@@ -69,11 +70,19 @@ export const styleSheet = createStyleSheet('MuiInput', (theme) => {
       },
     },
     disabled: {
-      color: theme.palette.text.disabled,
+      color: theme.palette.input.disabled,
       cursor: 'not-allowed',
     },
     underline: {
-      borderBottom: `1px solid ${theme.palette.text.divider}`,
+      borderBottom: `1px solid ${theme.palette.input.bottomLine}`,
+      '&:hover:not($disabled)': {
+        borderBottom: `2px solid ${theme.palette.text.primary}`,
+        marginBottom: -1,
+        transition: transitions.create('border-color', {
+          duration: transitions.duration.shorter,
+          easing: transitions.easing.ease,
+        }),
+      },
       '&$disabled': {
         borderBottomStyle: 'dotted',
```

Comment:
The dots looks too close, we could be using that trick (wrong color in the example):
```css
    background-image: linear-gradient(to right, black 33%, rgba(255,255,255,0) 0%);
    background-position: bottom;
    background-size: 3px 1px;
    background-repeat: repeat-x;
```